ansible/roles/csm.sbps.dns_srv_records/files/sbps_dns_srv_records.sh

```diff
@@ -42,12 +42,19 @@ system_name="$(cat /etc/environment | grep SYSTEM_NAME | awk -F= '{print $2;}')"
 while read -r line; do
   ncn_worker_node=`echo "$line" | awk -F ":" '{print $1}'`
   iscsi_server_id="id-$(echo $ncn_worker_node | awk -F "-" '{print $2}' | awk '{print substr($1,2);}')"
-  hsn_ip=`echo "$line" | awk -F ":" '{print $2}'`
+
+  hsn_ip=`echo "$line" | awk -F ":" '{print $2}'` || true
   nmn_ip=`echo "$line" | awk -F ":" '{print $3}'`
 
-  hsn_srv_records="$hsn_srv_records{\"content\": \"1 0 3260 iscsi-server-"${iscsi_server_id}.hsn.${system_name}".hpc.amslabs.hpecorp.net.\",\"disabled\": false},"
+  if [[ -n $hsn_ip ]]
+  then
+    hsn_srv_records="$hsn_srv_records{\"content\": \"1 0 3260 iscsi-server-"${iscsi_server_id}.hsn.${system_name}".hpc.amslabs.hpecorp.net.\",\"disabled\": false},"
+
+    hsn_a_records="$hsn_a_records{\"comments\": [], \"name\": \"iscsi-server-"${iscsi_server_id}.hsn.${system_name}".hpc.amslabs.hpecorp.net.\",\"changetype\":\"REPLACE\",\"records\":[{\"content\": \"${hsn_ip}\",\"disabled\": false}],\"ttl\": 3600,\"type\": \"A\"},"
```

Comment:
Hard-coded references to `hpc.amslabs.hpecorp.net` will break at customer sites.  You will need to dynamically discover the site's domain name, and use that instead of hardcoding it.